app/src/main/java/com/aaronjwood/portauthority/async/ScanHostsAsyncTask.java

```diff
@@ -146,13 +166,24 @@ protected final void onProgressUpdate(Void... params) {
 
         try {
             if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
-                Process ipProc = Runtime.getRuntime().exec(IP_CMD);
-                ipProc.waitFor();
-                if (ipProc.exitValue() != 0) {
+                int returnCode = 0;
```

Comment:
Could you split this out into a reusable method? I noticed it is the same as the diff above.